File: dualtech-ojt-portal/extract.py
import re

with open('public/tsdportal.html', 'r', encoding='utf-8') as f:
    content = f.read()

match = re.search(r'<script type="text/babel" data-type="module">(.*?)</script>', content, re.DOTALL)
if match:
    with open('tsd-portal/src/app_extracted.jsx', 'w', encoding='utf-8') as f:
        f.write(match.group(1).strip())
    print("Successfully extracted React code.")
else:
    print("Could not find the script tag.")

File: dualtech-ojt-portal/public/extract.js
const fs = require('fs');
const content = fs.readFileSync('c:/Users/rober/dualtech-ojt-portal/public/ic-management.html', 'utf8');
const scriptMatch = content.match(/<script type="text\/babel"[^>]*>([\s\S]*?)<\/script>/i);
if (scriptMatch) {
    fs.writeFileSync('temp_script.jsx', scriptMatch[1]);
    console.log('Extracted JSX to temp_script.jsx');
} else {
    console.log('Could not find script block');
}
